src/plugingen.nim

```diff
@@ -0,0 +1,109 @@
+import std/[macros, compilesettings]
+
+const
+  QtQmlCFlags =
+    gorge("pkg-config --cflags Qt6Qml") &
+    (when defined(gcc) or defined(llvm): " -fPIC" else: "")
+
+macro createDirAtCompileTime(dir: string): untyped =
+  result = quote do:
+    static:
+      discard staticExec("mkdir -p " & `dir`)
+
+macro writeFileAtCompileTime(filename: string, content: string): untyped =
+  result = quote do:
+    static:
+      writeFile(`filename`, `content`)
+
+
+func encodeCBORPluginMetaCxxArray(
+  iid, className, uri: static[string]
```

Comment:
these don't really have to be `static` as long as the whole thing is run in a static context